### DISCUSSION TITLE
multipart/form-data

### DISCUSSION THREAD

**Discussion by thebyrd** [encoredev/encore#44] [Help] [answered]
What is the best way to have a service accept a file as a parameter? 

The equivalent thing I am trying to do in a typical golang web app would be: https://tutorialedge.net/golang/go-file-upload-tutorial/

**Comment by eandre** [comment, marked as answer]
Hey @thebyrd, great question. The best way to do things like this is by dropping down an abstraction level and using **raw endpoints**:

```go
//encore:api public raw
func MyHandler(w http.ResponseWriter, req *http.Request) { /* ... */ }
```

From there on the TutorialEdge code should work completely unmodified.

**Reply by thebyrd** [reply]
thanks!

**Reply by DrSensor** [reply]
@eandre Is there a way to add some sort of annotation?
When I generate the client code from
```go
// Submit something
//encore:api public raw path=/rickroll
func Submit(w http.ResponseWriter, r *http.Request) {/*handle multipart data*/}
```
it give me
```ts
        /**
         * Submit something
         */
        public Submit(): Promise<void> {
            return this.baseClient.doVoid("POST", `/rickroll`)
        }
```
This is incorrect since I can't modify the request body

**Reply by eandre** [reply]
Hey @DrSensor, we’ve actually been working on a much improved client generator that will support this. It’s being released tomorrow most likely :)
